it/controllers/benefits/CompanyVanFuelBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,429 @@
+  import org.jsoup.Jsoup
+  import org.jsoup.nodes.Document
+  import play.api.http.HeaderNames
+  import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+  import play.api.libs.ws.WSResponse
+  import utils.{EmploymentDatabaseHelper, IntegrationTest, ViewHelpers}
+
+  class CompanyVanFuelBenefitsControllerISpec extends IntegrationTest with ViewHelpers with EmploymentDatabaseHelper {
+
+    val taxYearEOY: Int = taxYear-1
+    val employmentId: String = "001"
+
+    private val userRequest = User(mtditid, None, nino, sessionId, affinityGroup)(fakeRequest)
+
+    private def employmentUserData(isPrior: Boolean, employmentCyaModel: EmploymentCYAModel): EmploymentUserData =
+      EmploymentUserData(sessionId, mtditid, nino, taxYearEOY, employmentId, isPriorSubmission = isPrior, employmentCyaModel)
+
+    //def cyaModel(employerName: String, hmrc: Boolean, benefits: Option[BenefitsViewModel] = None): EmploymentCYAModel =
+    // EmploymentCYAModel(EmploymentDetails(employerName, currentDataIsHmrcHeld = hmrc), benefits)
+
+    def fullCarVanFuelModel: CarVanFuelModel =
+      CarVanFuelModel(
+        carVanFuelQuestion = Some(true),
+        carQuestion = Some(true),
+        car = Some(100),
+        carFuelQuestion = Some(true),
+        carFuel = Some(200),
+        vanQuestion = Some(true),
+        van = Some(300),
+        vanFuelQuestion = Some(true),
+        vanFuel = Some(400)
+      )
+
+    // models for Incomplete sections redirect tests
+    val benefitsWithNoBenefitsReceived: Option[BenefitsViewModel] = Some(BenefitsViewModel(isUsingCustomerData = true))
+
+    def benefitsWithNoCarVanFuelQuestion(carVanFuelQuestion: Option[Boolean] = Some(false)): Option[BenefitsViewModel] =
+      Some(BenefitsViewModel(isBenefitsReceived = true,
+        carVanFuelModel = Some(fullCarVanFuelModel.copy(carVanFuelQuestion = carVanFuelQuestion)),
+        isUsingCustomerData = true))
+
+    val benefitsWithEmptyCarQuestion: Option[BenefitsViewModel] =
+      Some(BenefitsViewModel(isBenefitsReceived = true,
+        carVanFuelModel = Some(fullCarVanFuelModel.copy(carQuestion = None)),
+        isUsingCustomerData = true))
+
+    val benefitsWithEmptyVanQuestion: Option[BenefitsViewModel] =
+      Some(BenefitsViewModel(isBenefitsReceived = true,
+        carVanFuelModel = Some(fullCarVanFuelModel.copy(vanQuestion = None)),
+        isUsingCustomerData = true))
+
+    val benefitsWithEmptyCarFuelQuestion: Option[BenefitsViewModel] =
+      Some(BenefitsViewModel(isBenefitsReceived = true,
+        carVanFuelModel = Some(fullCarVanFuelModel.copy(carFuelQuestion = None)),
+        isUsingCustomerData = true))
+
+    val benefitsCarQuestionYesNoAmount: Option[BenefitsViewModel] =
+      Some(BenefitsViewModel(isBenefitsReceived = true,
+        carVanFuelModel = Some(fullCarVanFuelModel.copy(car = None)), isUsingCustomerData = true))
+
+    val benefitsCarFuelQuestionYesNoAmount: Option[BenefitsViewModel] =
```

Comment:
Also can be removed 

(if you take the latest code I shared I already tidies these up)